src/app/services/map/map.service.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Option } from "@helpers/advancedTypes";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { ConfigService } from "@services/config/config.service";
+import { List } from "immutable";
+import { BehaviorSubject } from "rxjs";
+import { takeUntil } from "rxjs/operators";
+
+enum MapState {
+  loading = 0,
+  success = 1,
+  failure = 2,
+}
+
+/**
+ * Map Service for Google Maps Wrapper. This prevents the google maps
+ * api bundle being inserted into the application multiple times.
+ */
+@Injectable({ providedIn: "root" })
+export class MapService extends withUnsubscribe() {
+  /**
+   * Current state of google maps api bundle download (loading/success/failure)
+   */
+  public static mapState = MapState;
+  private static mapUrl = "https://maps.googleapis.com/maps/api/js";
+
+  /**
+   * Is google maps api bundle loaded
+   */
+  public isMapLoaded$ = new BehaviorSubject<MapState>(MapState.loading);
+
+  public constructor(private http: HttpClient, private config: ConfigService) {
+    super();
+    this.loadMap();
+  }
+
+  /**
+   * Is google maps api bundle loaded
+   */
+  public get isMapLoaded(): MapState {
+    return this.isMapLoaded$.getValue();
+  }
+
+  /**
+   * Determine if a map marker is valid
+   */
+  public isMarkerValid(marker: MapMarkerOption): boolean {
+    return (
+      typeof marker?.position?.lat === "number" &&
+      typeof marker.position.lng === "number"
```

Comment:
could add some bounds validations here too

- not null/undefined
- lat in [-90, 90]
- long in [-180, 180]